docs/core/deploying/native-aot/fixing-warnings.md

```diff
@@ -0,0 +1,80 @@
+
+```csharp
+Type t = typeof(int);
+while (true)
+{
+    t = typeof(GenericType<>).MakeGenericType(t);
+    Console.WriteLine(Activator.CreateInstance(t));
+}
+
+struct GenericType<T> { }
+```
+
+While the above program is not very useful, it represents an extreme case that requires an infinite number of generic types to be created when building the application as native AOT. Without native AOT, the program would run until it runs out of memory. With native AOT, we would not be able to even build it if we were to generate all the necessary types (the infinite number of them).
+
+In this case, native AOT build will issue following warning on the `MakeGenericType` line:
+
+```
+AOT analysis warning IL3050: Program.<Main>$(String[]): Using member 'System.Type.MakeGenericType(Type[])' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. The native code for this instantiation might not be available at runtime.
+```
+
+At run time, the application will indeed throw an exception from the `MakeGenericType` call.
+
+## Reacting to AOT warnings
+
+The AOT warnings are meant to bring predictability to native AOT builds. Majority of AOT warning are about possible run time exception in situations when native code wasn't generated to support the scenario. The broadest category is `RequiresDynamicCodeAttribute`.
+
+### RequiresDynamicCode
+
+<xref:System.Diagnostics.CodeAnalysis.RequiresDynamicCodeAttribute> is simple and broad: it's an attribute that means the member has been annotated incompatible with AOT, meaning that it might use reflection or some other mechanism to create new native code at runtime. This attribute is used when code is fundamentally not AOT compatible, or the native dependency is too complex to statically predict at build time. This would often be true for methods that use the Type.MakeGenericType API, reflection emit, or other runtime code generation technologies. An example would be:
+
+```csharp
+[RequiresDynamicCode("Use 'MethodFriendlyToAot' instead")]
+void MethodWithReflectionEmit() { ... }
+
+void TestMethod()
+{
+    // IL3050: Using method 'MethodWithReflectionEmit' which has 'RequiresDynamicCodeAttribute'
+    // can break functionality when AOT compiling. Use 'MethodFriendlyToAot' instead.
+    MethodWithReflectionEmit();
+}
+```
+
+There aren't many workarounds for `RequiresDynamicCode`. The best fix is to avoid calling the method at all when building as native AOT and use something else that's AOT-compatible. If you're writing a library and it's not in your control whether or not to call the method, you can also add `RequiresDynamicCode` to your own method. This will annotate your method as not AOT compatible. Adding `RequiresDynamicCode` will silence all AOT warnings in the given method, but will produce a warning whenever someone else calls it. For this reason, it is mostly useful to library authors to "bubble up" the warning to a public API.
```

Comment:
```suggestion
There aren't many workarounds for `RequiresDynamicCode`. The best fix is to avoid calling the method at all when building as native AOT and use something else that's AOT compatible. If you're writing a library and it's not in your control whether or not to call the method, you can also add `RequiresDynamicCode` to your own method. This will annotate your method as not AOT compatible. Adding `RequiresDynamicCode` silences all AOT warnings in the annotated method but will produce a warning whenever someone else calls it. For this reason, it's mostly useful to library authors to "bubble up" the warning to a public API.
```